src/auxiliary/math.jl

```diff
@@ -5,6 +5,12 @@
 @muladd begin
 #! format: noindent
 
+# `AbstractFloat` clashes with `ForwardDiff.Dual` => use `Real`
+sqrt_(x::T) where {T <: Real} = x < zero(x) ? oftype(x, NaN) : Base.sqrt(x)
+# < 0 suffices since log(0) = -Inf
+# `AbstractFloat` clashes with `ForwardDiff.Dual` => use `Real`
+log_(x::T) where {T <: Real} = x < zero(x) ? oftype(x, NaN) : Base.log(x)
+
```

Comment:
The idea would be that we should replace all `sqrt` and `log` with this, shouldn't we? Thus, they would be just `sqrt` and `log`.

Reply:
This should also get some decent comments why we don't use the base versions. I'm also not sure whether we should tag this as a breaking change - while it doesn't change anything that we have documented explicitly, it can change the behavior quite a bit.

Reply:
The reason why I have not replaced every `sqrt` with the domain checking version is that we have couple places where we take the `sqrt` out of a sum of squares (mostly l2 errors and some wave-speed estimates). In that cases the domain check is not really needed.

Reply:
But it makes everything much more complicated if you use `sqrt` somewhere and `sqrt_` in other places. I prefer having just one option everywhere.

Reply:
Done now except for the SIMD stuff mentioned in the comment below :+1: